uniffi_bindgen/src/bindings/python/templates/ErrorTemplate.py

```diff
@@ -0,0 +1,51 @@
+class RustError(ctypes.Structure):
+    _fields_ = [
+        ("code", ctypes.c_int32),
+        ("message", ctypes.c_char_p),
+    ]
+
+    def __str__(self):
+        return "RustError(code={}, message={})".format(self.code, str(self.message, "utf-8")) 
+
+{% for e in ci.iter_error_definitions() %}
+class {{e.name()}}:
+    {%- for value in e.values() %}
+    class {{value}}(Exception):
+        pass
+    {%- endfor %}
+
+    @staticmethod
+    def raise_err(err):
+        {%- for value in e.values() %}
+        if err.code == {{loop.index}}:
+            raise {{e.name()}}.{{value}}(err.__str__())
+        {% endfor %}
+        raise Exception("Unknown {{e.name()}} error code")
+{% endfor %}
+
+class RustErrorHelper:
+    def __init__(self):
+        self.err = RustError()
+
+    def __apply__(self, fn):
+        return fn(ctypes.byref(self.err))
+
+    def __raise__(self):
+    {%- for e in ci.iter_error_definitions() %}
+        if self.err.code == {{loop.index}}:
```

Comment:
Hmmm I'm not sure this does what you expect it to do.

Lets imagine this scenario:
```
[Error]
enum ArithmeticError {
"IntegerOverflow", "DivideByZero"
};

[Error]
enum StringError {
"EmptyStringError", "NullPointer"
};
```
If we receive an `ArithmeticError::DivideByZero` We would raise a `StringError::NullPointer`.

An alternative is to do what Swift does and accept an "empty" error of the correct type in `try_raise` (swift ref [here](https://github.com/mozilla/uniffi-rs/blob/bad6448e1f8e7ac80284f33197272ac833bbf39f/uniffi_bindgen/src/bindings/swift/templates/macros.swift#L10) notice how we pass an `{{e}}.NoError` to the `rustCall` helper).

It's not the best looking approach so feel free to try something else if you have another idea in mind 😄